third_party/terraform/tests/resource_compute_network_peering_test.go.erb

```diff
@@ -0,0 +1,163 @@
+
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccComputeNetworkPeeringDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeNetworkPeering_basic(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeNetworkPeeringExist("google_compute_network_peering.foo", &peering_beta),
+					testAccCheckComputeNetworkPeeringAutoCreateRoutes(true, &peering_beta),
+					<% unless version == 'ga' -%>
+					testAccCheckComputeNetworkPeeringImportCustomRoutes(false, &peering_beta),
+					testAccCheckComputeNetworkPeeringExportCustomRoutes(false, &peering_beta),
+					<% end -%>
+					testAccCheckComputeNetworkPeeringExist("google_compute_network_peering.bar", &peering_beta),
+					testAccCheckComputeNetworkPeeringAutoCreateRoutes(true, &peering_beta),
+					<% unless version == 'ga' -%>
+					testAccCheckComputeNetworkPeeringImportCustomRoutes(true, &peering_beta),
+					testAccCheckComputeNetworkPeeringExportCustomRoutes(true, &peering_beta),
+					<% end -%>
+				),
+			},
+		},
+	})
+
+}
+
+func testAccComputeNetworkPeeringDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_compute_network_peering" {
+			continue
+		}
+
+		_, err := config.clientComputeBeta.Networks.Get(
+			config.Project, rs.Primary.ID).Do()
+		if err == nil {
+			return fmt.Errorf("Network peering still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckComputeNetworkPeeringExist(n string, peering *computeBeta.NetworkPeering) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		config := testAccProvider.Meta().(*Config)
+
+		parts := strings.Split(rs.Primary.ID, "/")
+		if len(parts) != 2 {
+			return fmt.Errorf("Invalid network peering identifier: %s", rs.Primary.ID)
+		}
+
+		networkName, peeringName := parts[0], parts[1]
+
+		network, err := config.clientComputeBeta.Networks.Get(config.Project, networkName).Do()
+		if err != nil {
+			return err
+		}
+
+		found := findPeeringFromNetwork(network, peeringName)
+		if found == nil {
+			return fmt.Errorf("Network peering '%s' not found in network '%s'", peeringName, network.Name)
+		}
+		*peering = *found
+
+		return nil
+	}
+}
+
+func testAccCheckComputeNetworkPeeringAutoCreateRoutes(v bool, peering *computeBeta.NetworkPeering) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		if peering.ExchangeSubnetRoutes != v {
+			return fmt.Errorf("should ExchangeSubnetRouts set to %t if AutoCreateRoutes is set to %t", v, v)
+		}
+		return nil
+	}
+}
+
+func testAccCheckComputeNetworkPeeringImportCustomRoutes(v bool, peering *computeBeta.NetworkPeering) resource.TestCheckFunc {
```

Comment:
Oops, this and `Export` fail the linter in the ga version for being unused - will you wrap these in unless blocks as well?

Reply:
done